pkg/raft/raft_test.go

```diff
@@ -2178,82 +2178,100 @@ func TestLeaderIncreaseNext(t *testing.T) {
 	}
 }
 
-func TestSendAppendForProgressProbe(t *testing.T) {
-	testutils.RunTrueAndFalse(t, "store-liveness-enabled",
-		func(t *testing.T, storeLivenessEnabled bool) {
-			testOptions := emptyTestConfigModifierOpt()
-			if !storeLivenessEnabled {
-				// TODO(ibrahim): allow the test option to take a boolean to
-				// enable/disable fortification. This way we can refactor the tests and
-				// make them less verbose.
-				testOptions = withFortificationDisabled()
-			}
+func TestSendAppendForProgressProbeStoreLivenessDisabled(t *testing.T) {
+	r := newTestRaft(1, 10, 1, newTestMemoryStorage(withPeers(1, 2)),
+		withFortificationDisabled())
 
-			r := newTestRaft(1, 10, 1, newTestMemoryStorage(withPeers(1, 2)),
-				testOptions)
+	r.becomeCandidate()
+	r.becomeLeader()
 
-			r.becomeCandidate()
-			r.becomeLeader()
+	r.readMessages()
+	r.trk.Progress(2).BecomeProbe()
 
-			r.readMessages()
-			r.trk.Progress(2).BecomeProbe()
-
-			// each round is a heartbeat
-			for i := 0; i < 3; i++ {
-				if i == 0 {
-					// we expect that raft will only send out one msgAPP on the first
-					// loop. After that, the follower is paused until a heartbeat response
-					// is received.
-					mustAppendEntry(r, pb.Entry{Data: []byte("somedata")})
-					r.maybeSendAppend(2)
-					msg := r.readMessages()
-					assert.Len(t, msg, 1)
-					assert.Zero(t, msg[0].Index)
-				}
+	// each round is a heartbeat
+	for i := 0; i < 3; i++ {
+		if i == 0 {
+			// we expect that raft will only send out one msgAPP on the first
+			// loop. After that, the follower is paused until a heartbeat response
+			// is received.
+			mustAppendEntry(r, pb.Entry{Data: []byte("somedata")})
+			r.maybeSendAppend(2)
+			msg := r.readMessages()
+			assert.Len(t, msg, 1)
+			assert.Zero(t, msg[0].Index)
```

Comment:
What does this check do? Maybe check the `msg[0].Type` first, because the semantics of `Index` changes depending on the message type.

Reply:
Same in the other test.